main.py

```diff
@@ -57,63 +66,68 @@ def check(m):
+        
+    
+
 
     seconds = ("s", "sec", "secs", 'second', "seconds")
-    minutes = ("m", "min", "mins", "minute", "minutes")
-    hours = ("h", "hour", "hours")
+    minutes= ("m", "min", "mins", "minute", "minutes")
+    hours= ("h", "hour", "hours")
     days = ("d", "day", "days")
-    weeks = ("w", "week", "weeks")
     rawsince = since.content
-
+    
+    
     try:
         temp = re.compile("([0-9]+)([a-zA-Z]+)")
         if not temp.match(since.content):
             return await ctx.send("You did not specify a unit of time, please try again.")
+
         res = temp.match(since.content).groups()
         time = int(res[0])
         since = res[1]
-
+        
     except ValueError:
         return await ctx.send("You did not specify a unit of time, please try again.")
-
+        
     if since.lower() in seconds:
         timewait = time
     elif since.lower() in minutes:
-        timewait = time * 60
+        timewait = time*60
     elif since.lower() in hours:
-        timewait = time * 3600
+        timewait = time*3600
     elif since.lower() in days:
-        timewait = time * 86400
+        timewait = time*86400
     elif since.lower() in weeks:
-        timewait = time * 604800
+        timewait = time*604800
     else:
-
+        
         return await ctx.send("You did not specify a unit of time, please try again.")
-
+        
     await ctx.send("What would you like the prize to be?")
     try:
-        msg4 = await bot.wait_for('message', check=check, timeout=30.0)
+        msg4 = await bot.wait_for('message', check = check, timeout=30.0)
+
 
     except asyncio.TimeoutError:
         await ctx.send("You took to long, please try again.")
 
+
     logembed = discord.Embed(title="Giveaway Logged",
                              description=f"**Prize:** ``{msg4.content}``\n**Winners:** ``{winerscount}``\n**Channel:** {giveawaychannel.mention}\n**Host:** {ctx.author.mention}",
                              color=discord.Color.red())
     logembed.set_thumbnail(url=ctx.author.avatar_url)
 
-    logchannel = ctx.guild.get_channel(609431364445405194)  # Put your channel, you would like to send giveaway logs to.
+    guild = client.get_guild(796081913688883240)  # Put your guild ID here!
+    logchannel = guild.get_channel(799722523054833664)  # Put your channel, you would like to send giveaway logs to.
     await logchannel.send(embed=logembed)
 
     futuredate = datetime.utcnow() + timedelta(seconds=timewait)
-    embed1 = discord.Embed(color=discord.Color(random.randint(0x000000, 0xFFFFFF)),
-                           title=f"🎉GIVEAWAY🎉\n`{msg4.content}`", timestamp=futuredate,
-                           description=f'React with 🎉 to enter!\nHosted by: {ctx.author.mention}')
-
+    embed1 = discord.Embed(color = discord.Color(random.randint(0x000000, 0xFFFFFF)), title=f"🎉GIVEAWAY🎉\n`{msg4.content}`",timestamp = futuredate, description=f'React with 🎉 to enter!\nHosted by: {ctx.author.mention}')

```

Comment:
same code

Reply:
This is just what it looks like on mine, I did a pull request for additional print statements, has_permissions decorators and error decorators.

Reply:
> 
> 
> I added all those error messages, for missing permissions.
> ![image](https://user-images.githubusercontent.com/76411422/106044839-e43ceb00-60a5-11eb-8446-e8b0335e41cd.png)
> I also added error messages for commands that are upcoming.
> ![image](https://user-images.githubusercontent.com/76411422/106044903-f74fbb00-60a5-11eb-9e1c-255d4333b35b.png)
> If your questioning if I'm pulling my own weight, I think I am, I made an effort to create the terminal better then what it was just for aesthetic purposes, and I also took the time to create error messages for people who don't have the right permissions instead of the bot just returning nothing.
> 
> Think of it this way without my pull requests anybody can use the bot and start a giveaway, now they need permissions, and the bot will tell them if they don't. @helish88

I think manage_roles is enough

Reply:
I suggest theese commands have manage guild permissions needed to use it:
``!reroll``
``!pickwinner`` (Still being developed by me.)
I will change it.